src/mmtk.rs

```diff
@@ -597,3 +597,13 @@ impl<VM: VMBinding> MMTK<VM> {
             .initialize_object_metadata(object, false)
     }
 }
+
+#[no_mangle]
+pub fn mmtk_debug_print_object_info(object: crate::util::ObjectReference) {
+    println!(
+        "{}",
+        SFT_MAP
+            .get_checked(object.to_raw_address())
+            .debug_get_object_info(object)
+    )
```

Comment:
Instead of printing the result here, I suggest we let `debug_get_object_info` gradually print more and more information.  The reason is that when the user uses GDB or rr to debug a runtime, chance is high that the heap may have already been corrupted, and the metadata may have been inconsistent.  In this case, one of the steps inside `debug_get_object_info` may go wrong and panic before it returns a human-readable string.  Instead, if we let it gradually print more and more info, it can make the best effort even if it fails to finish normally

Reply:
Done. Now each function impl only fetches one piece of metadata at a time, and print one line at a time.